main/src/main/scala/sbt/Main.scala

```diff
@@ -141,12 +141,16 @@ object StandardMain {
   val console: ConsoleOut =
     ConsoleOut.systemOutOverwrite(ConsoleOut.overwriteContaining("Resolving "))
 
-  def initialGlobalLogging: GlobalLogging =
+  def initialGlobalLogging(file: File): GlobalLogging = {
+    Option(file).foreach(f => if (!f.exists()) IO.createDirectory(f))
     GlobalLogging.initial(
       MainAppender.globalDefault(console),
-      File.createTempFile("sbt", ".log"),
+      File.createTempFile("sbt-global-log", ".log", file),
       console
     )
+  }
+  @deprecated("use version that takes file argument", "1.4.0")
+  def initialGlobalLogging: GlobalLogging = initialGlobalLogging(null)
```

Comment:
No `null` please.

Reply:
In my opinion this is a good use of null because otherwise there has to be a variant of initialGlobalLogging that takes Option[File] as input.

Reply:
I'd much rather deal with `Some(...)` and `None.orNull` than `null`.